salt/_modules/metalk8s_cordon.py

```diff
@@ -99,4 +99,13 @@ def node_uncordon(node_name, **kwargs):
 
         salt '*' kubernetes.node_uncordon  node_name="minikube"
     '''
-    return node_set_unschedulable(node_name, False, **kwargs)
+    # need to remove all taints before node uncordon
+    set_no_taints = []
+    remove_all_taint = __salt__['metalk8s_kubernetes.node_set_taints'](
```

Comment:
Wait no, this would remove *all* taints on a node when uncordon'ing it? That ain't right at all!

Reply:
@NicolasT from the issue, it seems the node taint key in question happens to be `node.kubernetes.io/unschedulable`

Maybe I should specify the node taint to be removed?

Reply:
Nodes (can) have taints, for any key. We even rely on this for control-plane/infra/workload-plane separation, and other things.

Given this change, when uncordoning a node, all its taints would be removed, which is of course not what you intend to do: you're only looking to handle one taint. Can't remove any of the others.

Reply:
Thanks for the heads-up @NicolasT